### ISSUE TITLE
Resque Truck Factor

### ISSUE THREAD

**Issue by gavelino** [resque/resque#1317]
As part of my PhD research on code authorship, we calculated the Truck Factor (TF) of some popular GitHub repositories.

As you probably know, the [Truck (or Bus) Factor](https://en.wikipedia.org/wiki/Bus_factor) designates the minimal number of developers that have to be hit by a truck (or quit) before a project is incapacitated. In our work, we consider that a system is in trouble if more than 50% of its files become orphan (i.e., without a main author).

More details on our work in this preprint: https://peerj.com/preprints/1233

We calculated the TF for **Resque** and obtained a value of **6**.

The developers responsible for this TF are:

Chris Wanstrath - author of 23% of the files
Łukasz Strzałkowsk - author of 14% of the files
Ryan Biesemeyer - author of 12% of the files
Steve Klabnik - author of 11% of the files
Einar Jonsson - author of 8% of the files
Terence Lee - author of 8% of the files

To validate our results, we would like to ask **Resque** developers the following three brief questions:

(a) Do you agree that the listed developers are the main developers of **Resque**?

(b) Do you agree that **Resque** will be in trouble if the listed developers leave the project (e.g., if they win in the lottery, to be less morbid)?

(c) Does **Resque** have some characteristics that would attenuate the loss of the listed developers (e.g., detailed documentation)?

Thanks in advance for your collaboration,

Guilherme Avelino
PhD Student
Applied Software Engineering Group (ASERG)
UFMG, Brazil
http://aserg.labsoft.dcc.ufmg.br/

**Comment by yaauie** [comment]
Hello, Guilherme,

I'll do my best to answer your three questions:

(a) yes, that is an appropriate representation of authorship, if not entirely accurate percentage of ownership; for example: 
 - Chris Wanstrath, the principal author, he has not committed code to this project since 2011;
 - Terrance Lee helped address a lot of community input & bug reports in the early days, but hasn't committed since 2012;
 - Steve Klabnik was probably the closest that this project had to an active maintainer, but he too hasn't committed code since 2014;
 - Einar Jonsson and Łukasz Strzałkowsk both contributed massively in brief spikes in 2013, but I wouldn't consider them project maintainers; and
 - even through I have merged pull-requests recently, I have not authored any code in this project since 2014, and even that was only config/dependency bug fixes since 2013.

(b) none of the above developers are in active development of this project (myself included), and while a couple of us step in occasionally to merge pull-requests or solve an issue, most contribution comes in spurts from new consumers of the library; if this is a valid definition of "in trouble", then yes, the departure of the main code owners has put this project in peril.

(c) Resque has okay docs (if you look at the correct branch) and is relatively simple to get going, which I believe has mitigated the departure of active developers to an extent, but it Is hard for a project to thrive without active maintainers who have a vested interest in the project's stability.

I'm closing this issue, as I believe I have answered your questions.

Cheers, and good luck with your PhD studies.

-@yaauie
